File: CoolCommerce/ecommerce-frontend/src/domain/CheckoutInfo.ts
import Product from '@/domain/Product'
import {CustomerInfo} from '@/domain/CustomerInfo'
import {PaymentInfo} from '@/domain/PaymentInfo'

/**
 * Contains all information for a checkout
 */
export class CheckoutInfo {

    public customerInfo: CustomerInfo = new CustomerInfo()

    public paymentInfo: PaymentInfo = new PaymentInfo()

    public shippingOption = 'free'

    public cartItems: {product: Product, quantity: number, total: number}[] = []

}

File: CoolCommerce/ecommerce-main/src/main/groovy/com/coolcompany/ecommerce_main/api/domain/CheckoutInfo.java
package com.coolcompany.ecommerce_main.api.domain;

import com.coolcompany.ecommerce_payment.api.domain.CustomerInfo;
import com.coolcompany.ecommerce_payment.api.domain.PaymentInfo;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Created by Navíd Mitchell 🤪 on 1/13/23.
 */
@Data
@NoArgsConstructor
public class CheckoutInfo {

    private CustomerInfo customerInfo;

    private PaymentInfo paymentInfo;

    private String shippingOption;

    private List<CartItem> cartItems;

}
